marshal/tests/serialization/utils.go

```diff
@@ -0,0 +1,29 @@
+package serialization
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+)
+
+var UnmarshalErr = errors.New("unmarshal unexpectedly failed with error")
+var MarshalErr = errors.New("marshal unexpectedly failed with error")
+
+type UnequalError struct {
+	Expected string
+	Got      string
+}
+
+func (e UnequalError) Error() string {
+	return fmt.Sprintf("expect %s but got %s", e.Expected, e.Got)
+}
+
+func isCaseExpectedToFail(expected interface{}, types []reflect.Type) bool {
+	expectedT := reflect.TypeOf(expected)
+	for i := range types {
+		if types[i] == expectedT {
+			return true
+		}
+	}
+	return false
+}
```

Comment:
@illia-li  sorry for confusing you, but here is the deal:
When you move this function out of the `marshal.Set` it lost `Set` context, and because of that you need to change it's name to reflect a new meaning.

```suggestion
func isTypeOf(value interface{}, types []reflect.Type) bool {
	valueType := reflect.TypeOf(value)
	for i := range types {
		if types[i] == valueType {
			return true
		}
	}
	return false
}
```